clang/include/clang/Driver/Driver.h

```diff
@@ -658,6 +661,19 @@ class Driver {
   static void getDefaultModuleCachePath(SmallVectorImpl<char> &Result);
 
   bool getOffloadStaticLibSeen() const { return OffloadStaticLibSeen; };
+
+  /// addFPGATempDepFile - Add a file to be added to the bundling step of
+  /// an FPGA object.
+  void addFPGATempDepFile(const std::string &Name,
+                          const std::string &JA) const {
+    std::pair<std::string, std::string> TP(JA, Name);
+    FPGATempDepFiles.insert(TP);
+  }
+  /// getFPGATempDepFile - Get a file to be added to the bundling step of
+  /// an FPGA object.
+  const std::string getFPGATempDepFile(const std::string &JA) const {
+    return FPGATempDepFiles[JA];
```

Comment:
If I understand correctly, in case the entry for `JA` is absent, the value will be [default-constructed](https://llvm.org/doxygen/classllvm_1_1StringMap.html#a6814d65cea2c1d9da2892fd7e6385c5e). Should we somehow handle this at the caller site? Myself, I can't put my finger on some nasty edge case, but I'd appreciate your opinion on this.

Reply:
Should be handled at the caller site.  Should at least check for an empty string.

Reply:
Added check at caller.